info/v1/container.go

```diff
@@ -516,6 +516,31 @@ type FsStats struct {
 	WeightedIoTime uint64 `json:"weighted_io_time"`
 }
 
+// NvidiaGpuStats are measured using NVML.
+type NvidiaGpuStats struct {
+	// Minor number of the GPU device (/dev/nvidia[minor_number])
+	MinorNumber uint64 `json:"minor_number"`
+
+	// Product name of the GPU
+	Name string `json:"name"`
+
+	// Total installed GPU memory.
+	// unit: bytes
+	MemoryTotal uint64 `json:"memory_total"`
+
+	// Total GPU memory allocated.
+	// unit: bytes
+	MemoryUsed uint64 `json:"memory_used"`
+
+	// Percent of time over the past sample period (of nvml) during which
+	// one or more kernels was executing on the GPU.
+	Busy uint64 `json:"busy"`
```

Comment:
I am not familiar at all with nvml's capabilities, but I think it is generally preferable to use cumulative usage (in seconds), or another cumulative metric.  This allows metric aggregators to get average usage over short (for "instantaneous") or long (for average) intervals.  If we cant get that, then I would think it would be difficult to measure average usage over longer periods of time.  Do you know if the "sample period" is a constant?  The added c file below says "Each sample period may be between 1 second and 1/6 second, depending on the product being queried".

Reply:
Unfortunately, nvml doesn't expose cumulative usage. It only exposes this point-in-time gauge. The sample period is dependent upon the GPU type as you found out.